packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts

```diff
@@ -111,57 +112,97 @@ export function inlineEntityOnPluginEvent(event: PluginEvent, editor: IEditor) {
                     if (positionToUse) {
                         editor.select(positionToUse);
 
-                        editor.runAsync(asyncEditor => {
-                            if (isAfter) {
-                                const elAfter = asyncEditor.getElementAtCursor();
-                                removeDelimiterAttr(elAfter);
-                            }
-                            if (element) {
-                                removeNode(element);
-                            }
+                        editor.runAsync(aEditor => {
```

Comment:
this function is already too long (~140lines) let's break it into several pieces.